### ISSUE TITLE
Compile and return file path instead of raw CSS

### ISSUE THREAD

**Issue by NickStallman** [scssphp/scssphp#79]
Currently compile() returns raw CSS however in my application it would be beneficial for scssphp to store the CSS as a file, and just return the file path.

Basically the workflow I'm aiming for is:
- The page is rendered
- If a SCSS file is on the page, see if it's already cached or modified - this is already done in compile()
- If it needs recompiling then compile and store in a file
- Output the path to the compiled css in the <link> tag.
I'm essentially making a pure PHP on the fly Gulp replacement.

Also if a compileFile function was added then filemtime could be part of the Cache key meaning significantly improved performance rather than reading/loading any code.

All the key logic for this is already in scssphp and in the Cache class, but in the way it's currently implemented it's not possible since compile() only accepts code in and returns code.

I'm implementing these functions by extending the class, but this could be rather useful as part of the core library.

**Comment by robocoder** [comment]
Take a look at the `scssphp/server` repo. What you're proposing isn't within the scope of the library.

**Comment by NickStallman** [comment]
I did notice that, however in my particular use case trying to serve the CSS via PHP would be slower and slightly more difficult than my approach of compiling to a static file.

I also noticed some other small issues like the Cache class is quite a nice generic cache with dependencies, except cleanCache() is restricted to only cleaning hard coded scsscache files with no way to use the class to cache anything else.

I could do a couple of PR's for these without altering any existing behaviour at all?
